syllabus/core/stat_recorder.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+from syllabus.task_space import TaskSpace
+from gymnasium.spaces import Discrete #, MultiDiscrete?
+import json
+import os
+
+def convert_numpy(obj):
+    if isinstance(obj, np.generic):
+        return obj.item()  # Use .item() to convert numpy types to native Python types
+    raise TypeError
+
+class StatRecorder:
+    """
+    Individual stat tracking for each task.
+    """
+
+    def __init__(self, task_space: TaskSpace):
+        """Initialize the StatRecorder"""
+
+        self.write_path = '/Users/allisonyang/Downloads'
+
+        self.task_space = task_space
+
+        assert isinstance(self.task_space, TaskSpace), f"task_space must be a TaskSpace object. Got {type(task_space)} instead."
+        assert isinstance(self.task_space.gym_space, Discrete), f"assume that task_space should be of type Discrete"
+
+        self.tasks = self.task_space.get_tasks()
+        self.num_tasks = self.task_space.num_tasks
+
+        self.records = {task: [] for task in self.tasks}
+        self.stats = {task: {} for task in self.tasks}
```

Comment:
Instead of tracking the full list, track these efficiently. If we train for 10M episodes then it would be impossible to take the average of these lists. You can look up the running mean formulas, I think it's `average_mean = ((average_mean * N) + new_mean) / (N+1)` or something like that. There might be one for variance too

Reply:
Alternatively, it would be good to provide an option for only saving the past N episodes, so rather than taking the average over all of training, it's an average over the past N episodes. I think some normalization schemes prefer that method because returns change during training. It would be good to provide both options (let the user choose and configure each)